curvlinops/jacobian.py

```diff
@@ -7,30 +7,32 @@
 
 from backpack.hessianfree.lop import transposed_jacobian_vector_product as vjp
 from backpack.hessianfree.rop import jacobian_vector_product as jvp
-from numpy import allclose, ndarray
-from torch import Tensor, cat, from_numpy, no_grad, stack, zeros_like
+from torch import Tensor, cat, stack, zeros_like
 from torch.nn import Parameter
 
-from curvlinops._base import _LinearOperator
+from curvlinops._torch_base import CurvatureLinearOperator
 
 
-class JacobianLinearOperator(_LinearOperator):
-    """Linear operator for the Jacobian.
+class JacobianLinearOperator(CurvatureLinearOperator):
+    """Linear operator of the Jacobian.
 
-    Can be used with SciPy.
+    Attributes:
+        FIXED_DATA_ORDER: Whether the data order must be fix. ``True`` for Jacobians.
     """
 
+    FIXED_DATA_ORDER: bool = True
+
     def __init__(
         self,
-        model_func: Callable[[Tensor], Tensor],
+        model_func: Callable[[Union[MutableMapping, Tensor]], Tensor],
         params: List[Parameter],
         data: Iterable[Tuple[Union[Tensor, MutableMapping], Tensor]],
         progressbar: bool = False,
         check_deterministic: bool = True,
```

Comment:
I think the cost remains the same.

Before this PR, the deterministic checks were already checking the loss and gradient computed over the data set. This required a forward and backward pass through the net over all batches.

This PR only introduces new `allclose` checks (on the network's prediction, and if enabled also on the data), but the number of forward passes through the net remains the same.

You have a point that we should document more clearly a 'good practise' how to handle deterministic checks. I still think it is the best default to run them every time a new instance is created, but it would be great to make it easy for users to relax this, once they have confidence their linear operator is deterministic.

Reply:
Could you maybe create an issue for this?

Reply:
Oh of course. Sure, will create an issue.